system/jlib/jthread.cpp

```diff
@@ -1017,13 +1026,11 @@ class CThreadPool: public CThreadPoolBase, implements IThreadPool, public CInter
                         throw MakeStringException(0, "No threads available in pool %s", poolname.get());
                     IWARNLOG("Pool limit exceeded for %s", poolname.get());
```

Comment:
This code continues on with a new thread even when it timed out waiting for a free slot (when timeout == 0)
Is that what we want ?  Should it instead wait 'forever' for a free slot when timeout == 0 ?